File: frontend/src/api/complaint.ts
import { get, post, put } from './request'
import type { PageData } from '@/types/common'

export interface Complaint {
  id: number
  orderId: number
  complainantId: number
  respondentId: number
  type: number
  content: string
  evidenceImages?: string
  status: number
  arbitrationResult?: string
  arbitratorId?: number
  arbitratedAt?: string
  createdAt: string
  updatedAt: string
}

export function createComplaint(data: {
  orderId: number
  type: number
  content: string
  evidenceImages?: string
}) {
  return post<Complaint>('/complaints', data)
}

export function getComplaint(id: number) {
  return get<Complaint>(`/complaints/${id}`)
}

export function getMyComplaints(params?: any) {
  return get<PageData<Complaint>>('/complaints/my', params)
}

// Admin API
export function getAdminComplaints(params?: any) {
  return get<PageData<Complaint>>('/admin/complaints', params)
}

export function getAdminComplaint(id: number) {
  return get<Complaint>(`/admin/complaints/${id}`)
}

export function arbitrateComplaint(data: {
  complaintId: number
  arbitrationResult: string
  decision: number
}) {
  return put<Complaint>('/admin/complaints/arbitrate', data)
}

File: backend/src/main/java/com/dianjing/entity/Complaint.java
package com.dianjing.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "complaints")
public class Complaint {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "order_id", nullable = false)
    private Long orderId;

    @Column(name = "complainant_id", nullable = false)
    private Long complainantId;

    @Column(name = "respondent_id", nullable = false)
    private Long respondentId;

    @Column(nullable = false)
    private Integer type;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @Column(name = "evidence_images", columnDefinition = "TEXT")
    private String evidenceImages;

    @Column(nullable = false)
    private Integer status = 0;

    @Column(name = "arbitration_result", columnDefinition = "TEXT")
    private String arbitrationResult;

    @Column(name = "arbitrator_id")
    private Long arbitratorId;

    @Column(name = "arbitrated_at")
    private LocalDateTime arbitratedAt;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
